syntax = "proto3";

package inventory;

service InventoryService {
  rpc GetInventory (GetInventoryRequest) returns (InventoryResponse);
  rpc UpdateInventory (UpdateInventoryRequest) returns (InventoryResponse);
  rpc GetInventories (GetInventoriesRequest) returns (InventoriesResponse);
}

message Inventory {
  int32 product_id = 1;
  int32 quantity = 2;
  int64 updated_at = 3;
}

message GetInventoryRequest {
  int32 product_id = 1;
}

message UpdateInventoryRequest {
  int32 product_id = 1;
  int32 quantity = 2;
}

message GetInventoriesRequest {
  repeated int32 product_ids = 1;
}

message InventoryResponse {
  bool success = 1;
  string message = 2;
  Inventory inventory = 3;
}

message InventoriesResponse {
  repeated Inventory inventories = 1;
}

